sources/web/datalab/polymer/components/datalab-files/datalab-files.ts

```diff
@@ -598,6 +601,46 @@ class FilesElement extends Polymer.Element {
     }
   }
 
+  _toggleAltAddToolbar() {
+    this.$.altAddToolbar.toggle();
+  }
+
+  _toggleAltUpdateToolbar() {
+    this.$.altUpdateToolbar.toggle();
+  }
+
+  /**
+   * Called on window.resize, collapses elements to keep the element usable
+   * on small screens.
+   */
+  _resizeHandler() {
+    const width = this.$.toolbar.clientWidth;
+    // Collapse the add buttons on the toolbar
+    if (width < this._addToolbarCollapseThreshold) {
+      Utils.moveElementChildren(this.$.addToolbar, this.$.altAddToolbar);
+      this.$.altAddToolbarToggle.style.display = "inline-flex";
+    } else {
+      Utils.moveElementChildren(this.$.altAddToolbar, this.$.addToolbar);
+      this.$.altAddToolbarToggle.style.display = "none";
+      this.$.altAddToolbar.close();
+    }
+
+    // Collapse the file operations buttons on the toolbar
+    if (width < this._updateToolbarCollapseThreshold) {
```

Comment:
I agree about the general concept of not repeating code of course, but this case looks too specialized for a function, even just thinking about that function's parameters makes me want to redesign the whole thing.
I think I'll keep this for now, we can rethink it if we're touching this code again.